dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -115,6 +118,17 @@ public Response execute(Endpoint endpoint, Request request) throws IOException {
             // not retried by default, so ours implements SafeLoggable and retains the simple-name for
             // cleaner metrics.
             throw new SafeConnectTimeoutException(e, failureDiagnosticArgs(endpoint, request, startTime));
+        } catch (NoHttpResponseException e) {
+            // This should not be retried on non-idempotent endpoints since it can happen when the server
+            // already processed the request, however this can also happen if a persistent connection is timed out and
+            // closed by the server or intermediate proxy, in that case it's important that we do retry. Therefore, we
+            // only allow retries if less than 5 seconds have passed.
+            if (Duration.ofNanos(System.nanoTime() - startTime).getSeconds() <= 5) {
```

Comment:
Reading a bit more into this: https://github.com/palantir/dialogue/blob/develop/dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java#L410 sets our inactivity check value. If anything we should probably tie the value to this.

So apache ends up calling BHttpConnection#isStale which tries to do a read. So we're worried about the race condition between that check and finishing sending the response? How often do we hit this condition? What if we just not retry at all, regardless of whether the request is retryable?